server/plugin.ts

```diff
@@ -57,9 +67,14 @@ export class FlowFrameworkDashboardsPlugin
       }
     );
 
-    const opensearchRoutesService = new OpenSearchRoutesService(client);
-    const flowFrameworkRoutesService = new FlowFrameworkRoutesService(client);
-    const mlRoutesService = new MLRoutesService(client);
+    const dataSourceEnabled = !!dataSource;
+    if (dataSourceEnabled) {
+      dataSource.registerCustomApiSchema(flowFrameworkPlugin);
+      dataSource.registerCustomApiSchema(mlPlugin);
```

Comment:
what does this do?

Reply:
During plugin set up, in order to register the plugin API schema/spec, we do the following dataSource.registerCustomApiSchema(flowFrameworkPlugin)


Reference: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7578#issue-2438193350